src/solver/conv_asm_implicit_gemm_gtc_bwd_nhwc.cpp

```diff
@@ -946,16 +972,57 @@ ConvSolution ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC::GetSolution(
     if(isGfx90aFp16altSupport)
         GenerateClangDefsym(opts_0, "igemm_bwd_fp16_alt_impl", 0);
     result.construction_params[0].comp_options = opts_0.str();
+    std::ostringstream msg;
 
     if(isGfx90aFp16altSupport)
     {
         result.construction_params.push_back(kernel);
         std::ostringstream opts_1(options.str(), std::ios_base::ate);
         GenerateClangDefsym(opts_1, "igemm_bwd_fp16_alt_impl", 1);
         result.construction_params[1].comp_options = opts_1.str();
+        msg << ", fp16_alt:" << ctx.conv_problem.GetConv().attribute.gfx90aFp16alt.GetBwd();
+    }
+
+    if(isNCHW)
+    {
+        const auto& hi    = ctx.out_height;
+        const auto& wi    = ctx.out_width;
+        const auto& n     = ctx.batch_sz;
+        const auto& k     = ctx.n_inputs;
+        const auto& c     = ctx.n_outputs;
+        const auto& ho    = ctx.in_height;
+        const auto& wo    = ctx.in_width;
+        const auto& y     = ctx.kernel_size_h;
+        const auto& x     = ctx.kernel_size_w;
+        const auto& group = ctx.group_counts;
+
+        TransposeSolution_NHWC2NCHW trans_input(ctx, ctx.out_data_type, n, c, hi, wi);
+        TransposeSolution_NCHW2NHWC trans_weight(ctx,
+                                                 ctx.weights_data_type,
+                                                 k,
+                                                 c / group,
+                                                 y,
+                                                 x); // group * k_per_group as batch for weight
+        TransposeSolution_NCHW2NHWC trans_output(ctx, ctx.in_data_type, n, k, ho, wo);
+
+        if(!trans_input.IsSkippable())
+        {
+            result.construction_params.push_back(trans_input.GetKernel());
+            msg << ", inp trans:" << trans_input.GetKernelName();
+        }
+        if(!trans_weight.IsSkippable())
+        {
+            result.construction_params.push_back(trans_weight.GetKernel());
+            msg << ", wei trans:" << trans_weight.GetKernelName();
+        }
+        if(!trans_output.IsSkippable())
+        {
+            result.construction_params.push_back(trans_output.GetKernel());
+            msg << ", out trans:" << trans_output.GetKernelName();
+        }
     }
 
-    MIOPEN_LOG_I2("ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC: " + config.ToString());
+    MIOPEN_LOG_I2("ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC: " + config.ToString() + msg.str());
```

Comment:
You can use ostringstream syntax in `MIOPEN_LOG*` macros.
```suggestion
    MIOPEN_LOG_I2("ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC: " << config.ToString() << msg.str());
```

Reply:
Logging output begins with `[GetSolution]`, which creates an impression that logging happens during `GetSolution()` call. This is not so, however. The logging happens during invocation of kernels. Let's clearly indicate this:
```suggestion
    MIOPEN_LOG_I2("[INVOKER] ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC: " + config.ToString() + msg.str());
```

Reply:
Please always use SolverDbId():
```suggestion
    MIOPEN_LOG_I2(SolverDbId(*this) << ": " ...);
```

Reply:
> Logging output begins with `[GetSolution]`, which creates an impression that logging happens during `GetSolution()` call. This is not so, however. The logging happens during invocation of kernels. Let's clearly indicate this:

This IS in GetSolution right? not actually invoker the kernels

Reply:
Invoker is a lambda produced (instantiated) by another lambda (InvokerFactory). Both can be defined directly in GetSolution (like in this case) or somewhere else (only to avoid duplication of code).

InvokerFactory instance resides in the Solution object. The library runs it only when necessary, i.e. sometime later, after GetSolution(). The resulting Invoker object is instantiated and stored in the Invoker Cache.

Sometime more later, when the library wants to start the kernels, it will run Invoker.